woocommerce/woocommerce-gateway-stripe
# Hide Express Checkout in block editor if disabled

Fixes #3437

## Changes proposed in this Pull Request:
Do not display the express checkout buttons (PRB or ECE) in the block editor when they are not enabled as payment methods.







## Testing instructions
1. With `Apple Pay/Google Pay` and/or `Link` enabled, go to `wp-admin` Pages and edit the (Block) Checkout page.
2. You should see the Express Payment block in the editor. See Screenshot 1.
3. Disable Apple Pay/Google Pay, and refresh the editor.
4. You should no long see the Express Payment block in the editor. See Screenshot 2.
5. Test this for both ECE and PRB by toggling the feature flag.
   - To enable ECE, add `_wcstripe_feature_ece` with the value of `yes` to your `wp_options` table. 
6. Test the same for when editing the Cart page.
7. Verify that the actual cart and checkout pages still work as expected, i.e. customer can see Apple Pay/Google Pay if enabled, and can purchase using them.



**Screenshot 1. Apple Pay/Google Pay is enabled, and appears in the block editor.**
<img width="1728" alt="Screenshot 2024-10-18 at 3 57 58 PM" src="https://github.com/user-attachments/assets/6c8a76b3-5303-4028-9dc4-a9db1c582a9f">


**Screenshot 2: Apple Pay/Google Pay is not enabled, and does not appear in the block editor.**
<img width="1728" alt="Screenshot 2024-10-18 at 3 58 14 PM" src="https://github.com/user-attachments/assets/b015b3a5-8093-4be8-b953-e600a16af757">

    




---


**Post merge**
